lightning/src/ln/peers/conduit.rs

```diff
@@ -196,6 +196,16 @@ impl Decryptor {
 	fn increment_nonce(&mut self) {
 		Conduit::increment_nonce(&mut self.receiving_nonce, &mut self.receiving_chaining_key, &mut self.receiving_key);
 	}
+
+	// Used in tests to determine whether or not excess bytes entered the conduit without needing to bring up
+	// infrastructure to properly encode it
+	#[cfg(test)]
+	pub fn read_buffer_length(&self) -> usize {
+		match &self.read_buffer {
+			&Some(ref vec) => { vec.len() }
+			&None => 0
+		}
+	}
```

Comment:
Consider having an assert against the content rather than simply exposing the length.

Reply:
This was intentional. My test philosophy is to not add test-assert methods as extensions of existing classes. Those are used primarily in test-only Mock objects. Exposing internal lengths isn't great, but it is better than not having test coverage at all. Another great place for this project have a common pattern or doc that you can point new developers towards. Otherwise, the reviews seem pretty disjoint depending on which reviewer looks at the code and their own personal style.

I agree that this pattern definitely points to a poor design choice in 494 to have the Conduit(Encryptor/Decryptor) created and instantiated as part of the handshake code. This came up in one of Matt's comments later on as well.

An alternative is to have the handshake code return the number of bytes consumed and the higher layers can instantiate the Conduit after the handshake is complete. This would allow the tests to verify the number of bytes read matches expectations and removes this test-only method.

This wasn't done in the original patches due to the existing layering pre-Transport. The PeerManager code handling the number of bytes read as well as the completed handshake state made the code hard to reason about and maintain. This may be much simpler now that more distinct layers exist.

In many of the changes I made here, I chose to keep the existing design choices from 494 to limit the exposed surface for feedback and iteration. I was under the assumption that 494 was reviewed from a design perspective and just needed to be cleaned up a bit. Obviously, that wasn't that case but should explain some of the confusing choices that were made in these commits.